### PR TITLE
[Merged by Bors] - treewide cleanups and refactoring for initial tests

### PR DESCRIPTION
**Pull request by nrdxp**
- [x] refactor lib into separate files, similar to NixOS/nixpkgs/lib.
- [x] refactor ci to automatically generate derivations from flake outputs
- [x] remove cluttered indirection statements throughout the codebase
- [x] refactor hosts to allow for upcoming integration tests
- [x] improve ambiguity in the existing docs 
- [x] add [BORS](https://bors.tech) support
- [x] add initial integration test
- [x] write tests documentation
- [x] test lib
- [x] improve version string generation, and do so automatically for pkgs/flake.nix sources

Clean up the codebase as best we can in preparation for #152 and add tests. From now on, all PRs will be merged with BORS.

### CODE REVIEW

**Comment by nrdxp** [comment]
bors ping

**Comment by nrdxp** [comment]
bors try

**Comment by nrdxp** [comment]
bors try

**Review by Pacman99 — APPROVED** [review]

File: lib/devos/devosSystem.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+            ({ config, ... }: {
+              isoImage.isoBaseName = "nixos-" + config.networking.hostName;
+              # confilcts with networking.wireless which might be slightly
+              # more useful on a stick
+              networking.networkmanager.enable = lib.mkForce false;
+              # confilcts with networking.wireless
+              networking.wireless.iwd.enable = lib.mkForce false;
+            })
+          ];
+        })).config;
```
Comment (Pacman99) [inline lib/devos/devosSystem.nix:42]: Unrelated, I can submit a seperate PR for this. It turns out home-manager.useUserPackages has to be set to false with home-manager only configurations, so I added a `hmConfig` for hm-only configs.
```
hmConfig = (nixosSystem
            (args // {
              modules = modules ++ [
                ({ ... }: {
                  home-manager.useUserPackages = mkForce false;
                })
              ];
            })).config;
```

**Review by blaggacao — APPROVED** [review]
Lib looks a lot organized and becomes easier to reason about.

**Review by Pacman99 — APPROVED** [review]

**Pull request closed without merging**